types/AbstractBaseType.php

```diff
@@ -451,6 +451,17 @@ public function rawValue($value) {
     }
 
     /**
+     * This is called when a single string is needed to represent this Type's current
+     * value as a single (non-HTML) string. Eg. in a dropdown or in autocompletion.
+     *
+     * @param string $value
+     * @return string
+     */
+    public function displayValue($value) {
```

Comment:
I'm not sure I understand your issue.

* value - whatever was *selected* from the database (eg. a complex JSON string)
* rawValue - whatever is *stored* in the database column in the data_* table
* displayValue - a single, non-HTML string derived from the value

For most fields all three values are exactly the same. For others they are not. Eg. Page with Titles.